libs/geospatial/src/main/java/org/codice/ddf/libs/geo/util/GeospatialUtil.java

```diff
@@ -263,4 +274,66 @@ public static Geometry transformToEPSG4326LonLatFormat(
     }
     return transformedGeometry;
   }
+
+  /**
+   * Create a circular polygon from a lat, lon, radius, vertices, and distanceTolerance.
+   *
+   * @param lat - latitude in EPSG:4326 decimal degrees
+   * @param lon - longitude in EPSG:4326 decimal degrees
+   * @return radius - distance from the center point in meters
+   * @return maxVertices - the maximum number of vertices in the finished circle polygon
+   * @return distanceTolerance - the maximum distance from the original vertices a reduced vertex
+   *     may lie on a simplified circular polygon
+   */
+  public static Geometry createCirclePolygon(
+      double lat, double lon, double radius, int maxVertices, double distanceTolerance) {
+    double step = distanceTolerance;
+    Measure measure = Measure.valueOf(radius, SI.METER);
+    Point jtsPoint = new GeometryFactory().createPoint(new Coordinate(lon, lat));
+
+    Geometry bufferedCircle =
+        createBufferedCircleFromPoint(measure, DefaultGeographicCRS.WGS84, jtsPoint);
+    Geometry simplifiedCircle =
+        createBufferedCircleFromPoint(measure, DefaultGeographicCRS.WGS84, jtsPoint);
+
+    int maxVerticesWithClosedPoint = maxVertices + 1;
+
+    while (simplifiedCircle.getCoordinates().length > maxVerticesWithClosedPoint) {
```

Comment:
From what I've seen, the first iteration is typically 32 vertices. The logic was suggested by @bdeining for the reduction. The idea here is to make a less and less approximation until there are an appropriate number of vertices.

Reply:
so why is it 32? if that's how the JTS library work, should put a note/comment then
What if the `numMultiPointRadiusVertices` is greater than 32, which is probably and overkill, but then we will never achieve the desired number of vertices as configured.

Reply:
good point, I'll define a max number of vertices same as I did with a min number and alert the user if the admin value is too high.

Reply:
one way to avoid the while may be to create a circle from the point (instead of using buffer) and then call linearize()

http://docs.geotools.org/latest/userguide/library/jts/geometry.html `Creating CircularString`

Reply:
:nit consider renaming `createBufferedCircleFromPoint` to `createBufferedPoint`

Reply:
never mind... linearize() works the same way - I thought you may be able to specify the number of points